src/Fabulous/View.fs

```diff
@@ -0,0 +1,41 @@
+namespace Fabulous
+
+module ViewHelpers =
+    let canReuseView (prevWidget: Widget) (currWidget: Widget) =
+        let prevKey = prevWidget.Key
+
+        if not(prevKey = currWidget.Key) then
+            false
+        else if (prevKey = Memo.MemoWidgetKey) then
+            Memo.canReuseMemoizedViewNode prevWidget currWidget
+        else
+            true
+
+module View =
+    let memo<'msg, 'key, 'marker when 'key: equality>
+        (key: 'key)
+        (fn: 'key -> WidgetBuilder<'msg, 'marker>)
+        : WidgetBuilder<'msg, Memo.Memoized<'marker>> =
+
+        let memo: Memo.MemoData =
+            {
+                KeyData = box key
+                KeyComparer = fun (prev: obj) (next: obj) -> unbox<'key> prev = unbox<'key> next
+                CreateWidget = fun k -> (fn(unbox<'key> k)).Compile()
+                KeyTypeHash = typeof<'key>.GetHashCode ()
```

Comment:
I think storing the `Type` directly is ok. From what I've tried a while back, there's no allocation for it

Reply:
yes, I was thinking about micro optimizing GC traversal (no need to traverse pointers to type there), but I think you are right about using types directly